pallets/xyk/src/lib.rs

```diff
@@ -503,14 +542,31 @@ impl<T: Trait> Module<T> {
 
         result
             .saturated_into::<u128>()
-            .saturated_into::<T::Balance>()
+            .saturated_into::<BalanceOf<T>>()
+    }
+
+    // TODO: remove option from function signature
+    pub fn get_next_asset_id() -> CurrencyIdOf<T> {
+        let currency_id = <NextCurrencyId<T>>::get();
+
+        <NextCurrencyId<T>>::mutate(|id| *id += One::one());
```

Comment:
getter should not mutate the value

Reply:
function was moved and renamed to 'generate_'